pkg/vmprovider/providers/vsphere/virtualmachine/backup_test.go

```diff
@@ -46,54 +47,67 @@ func backupTests() {
 		ctx = nil
 	})
 
-	Context("VM bootstrap data is NOT provided", func() {
+	Context("VM Kube data", func() {
+
+		BeforeEach(func() {
+			vmCtx.VM = builder.DummyVirtualMachine()
+		})
 
-		It("Should back up only VM kube data in ExtraConfig", func() {
+		It("Should backup VM kube data YAML without status field in ExtraConfig", func() {
 			Expect(virtualmachine.BackupVirtualMachine(vmCtx, vcVM, nil)).To(Succeed())
-			verifyBackupDataInExtraConfig(ctx, vcVM, vmCtx, nil)
+
+			vmCopy := vmCtx.VM.DeepCopy()
+			vmCopy.Status = vmopv1.VirtualMachineStatus{}
+			vmCopyYaml, err := yaml.Marshal(vmCopy)
+			Expect(err).NotTo(HaveOccurred())
+			expectedVal, err := util.EncodeGzipBase64(string(vmCopyYaml))
+			Expect(err).NotTo(HaveOccurred())
+			verifyBackupDataInExtraConfig(ctx, vcVM, constants.BackupVMKubeDataExtraConfigKey, expectedVal)
 		})
 	})
 
-	Context("VM bootstrap data is provided", func() {
+	Context("VM bootstrap data", func() {
 
-		BeforeEach(func() {
-			bootstrapData = map[string]string{"foo": "bar"}
-		})
+		It("Should back up bootstrap data as JSON in ExtraConfig", func() {
+			bootstrapData := map[string]string{"foo": "bar"}
+			Expect(virtualmachine.BackupVirtualMachine(vmCtx, vcVM, bootstrapData)).To(Succeed())
 
-		AfterEach(func() {
-			bootstrapData = nil
+			bootstrapDataJSON, err := json.Marshal(bootstrapData)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(bootstrapDataJSON).NotTo(BeEmpty())
+			expectedVal, err := util.EncodeGzipBase64(string(bootstrapDataJSON))
+			Expect(err).NotTo(HaveOccurred())
+			verifyBackupDataInExtraConfig(ctx, vcVM, constants.BackupVMBootstrapDataExtraConfigKey, expectedVal)
 		})
+	})
 
-		It("Should back up both VM kube data and bootstrap data in ExtraConfig", func() {
-			Expect(virtualmachine.BackupVirtualMachine(vmCtx, vcVM, bootstrapData)).To(Succeed())
-			verifyBackupDataInExtraConfig(ctx, vcVM, vmCtx, bootstrapData)
+	Context("VM Disk data", func() {
+
+		It("Should backup VM disk data as JSON in ExtraConfig", func() {
+			Expect(virtualmachine.BackupVirtualMachine(vmCtx, vcVM, nil)).To(Succeed())
+
+			// Use the default disk info from the vcSim VM for testing.
+			diskData := []virtualmachine.VMDiskData{
```

Comment:
Can we flip how this assertion is done around:

- assert the expected key is in the EC
- decode/uncompress the value
- assert expected size/value in the array

so it is clearer when something doesn't match what part that is. We have some helpers to convert the EC array to a map for easier lookup.

Reply:
Updated the `verifyBackupDataInExtraConfig()` using the suggested assertion.